src/flow.jl

```diff
@@ -283,3 +415,423 @@ end
+    # Edge 2 => 1
+    # Edge 3 => 2
+    # Edge 4 => 9
+    # ⋮ )
+    riverlength_bc = get(config.model, "riverlength_bc", 1.0e5) # river length at boundary point (ghost point)
+    alpha = get(config.model, "inertial_flow_alpha", 0.7) # stability coefficient for model time step (0.2-0.7)
+    h_thresh = get(config.model, "h_thresh", 1.0e-03) # depth threshold for flow at link
+    froude_limit = get(config.model, "froude_limit", true) # limit flow to subcritical according to Froude number
+
+    bankfull_elevation_2d = ncread(
+        nc,
+        param(config, "input.lateral.river.bankfull_elevation");
+        type = Float,
+        fill = 0,
+    )
+    bankfull_depth_2d = ncread(
+        nc,
+        param(config, "input.lateral.river.bankfull_depth");
+        type = Float,
+        fill = 0,
+    )
+    bankfull_depth = bankfull_depth_2d[inds]
+    zb = bankfull_elevation_2d[inds] - bankfull_depth # river bed elevation
+
+    bankfull_volume = bankfull_depth .* width .* dl
+
+    n_river = ncread(
+        nc,
+        param(config, "input.lateral.river.n", nothing);
+        sel = inds,
+        defaults = 0.036,
+        type = Float,
+    )
+
+    n = length(inds)
+    # set ghost points for boundary condition (downstream river outlet): river width and
+    # manning n is copied from the upstream cell, river elevation and h are set at 0.0 (sea
+    # level). river length at boundary point is by default 1.0e5 m (riverlength_bc).
+    index_pit = findall(x -> x == 5, ldd)
+    npits = length(index_pit)
+    add_vertex_edge_graph!(graph, index_pit)
+    append!(zb, fill(Float(0), npits))
+    append!(dl, fill(riverlength_bc, npits))
+    append!(width, width[index_pit])
+    append!(n_river, n_river[index_pit])
+
+    # for each link the src and dst node is required
+    nodes_at_link = adjacent_nodes_at_link(graph)
+    _ne = ne(graph)
+
+    # determine z, width, length and manning's n at links
+    zb_max = fill(Float(0), _ne)
+    width_at_link = fill(Float(0), _ne)
+    length_at_link = fill(Float(0), _ne)
+    mannings_n = fill(Float(0), _ne)
+    for i = 1:_ne
+        zb_max[i] = max(zb[nodes_at_link.src[i]], zb[nodes_at_link.dst[i]])
+        width_at_link[i] = min(width[nodes_at_link.dst[i]], width[nodes_at_link.src[i]])
+        length_at_link[i] = 0.5 * (dl[nodes_at_link.dst[i]] + dl[nodes_at_link.src[i]])
+        mannings_n[i] =
+            (
+                n_river[nodes_at_link.dst[i]] * dl[nodes_at_link.dst[i]] +
+                n_river[nodes_at_link.src[i]] * dl[nodes_at_link.src[i]]
+            ) / (dl[nodes_at_link.dst[i]] + dl[nodes_at_link.src[i]])
+    end
+
+    # set depth h of reservoir and lake location to bankfull depth
+    h = fill(0.0, n + length(index_pit))
+    if !isnothing(reservoir)
+        inds_reservoir = findall(x -> x > 0, reservoir_index)
+        h[inds_reservoir] = bankfull_depth[inds_reservoir]
+    end
+    if !isnothing(lake)
+        inds_lake = findall(x -> x > 0, lake_index)
```

Comment:
Nit, but FYI, this also works: `findall(>(0), lake_index)`, because `>(0)` creates a 1 argument function.

Reply:
Ah, ok, did change it to `>(0)`.